Add msghistory option (#16048)
vim/vim: runtime/doc/message.txt

				Clear messages, keeping only the {count} most
				recent ones.

> The number of remembered messages depends on 'maxmsghistory'.

Reviewer: I think it's better:
```suggestion
The number of remembered messages is determined by 'msghistory'.
```
